packages/common/object-map.js

```diff
@@ -33,17 +59,18 @@ const { entries, fromEntries } = Object;
  * if all the mapped values are Passable, then the returned object will be
  * a CopyRecord.
  *
- * @template {Record<string, any>} O
+ * @template {Record<string, unknown>} O
  * @template R map result
  * @param {O} original
- * @param {(value: O[keyof O], key: keyof O) => R} mapFn
- * @returns {Record<keyof O, R>}
+ * @param {<K extends keyof O>(value: O[K], key: K) => R} mapFn
+ * @returns {{ [K in keyof O]: K extends string ? R : never }}
  */
 export const objectMap = (original, mapFn) => {
-  const ents = entries(original);
-  const mapEnts = ents.map(
-    ([k, v]) => /** @type {[keyof O, R]} */ ([k, mapFn(v, k)]),
-  );
-  return /** @type {Record<keyof O, R>} */ (harden(fromEntries(mapEnts)));
+  const oldEntries = typedEntries(original);
+  /** @type {<K extends keyof O>(entry: [K, O[K]]) => [K, R]} */
+  const mapEntry = ([k, v]) => [k, mapFn(v, k)];
+  const newEntries = typedMap(oldEntries, mapEntry);
+  const newObj = fromTypedEntries(newEntries);
+  return /** @type {any} */ (harden(newObj));
```

Comment:
Where did it stick here?

Reply:
I'm not sure; it seems to have just gotten super confused:
```
Type '{ [Entry in [keyof O, R] as Entry[0]]: Entry[1]; }' is not assignable to type '{ [K in keyof O]: K extends string ? R : never; }'.
  Type 'keyof O' is not assignable to type '[keyof O, R]'.
    Type 'string | number | symbol' is not assignable to type '[keyof O, R]'.
      Type 'string' is not assignable to type '[keyof O, R]'.(2322)
```

Reply:
Looks like the type mapping failing ...

Reply:
Possibly https://github.com/microsoft/TypeScript/issues/49653#issuecomment-1172537589 ?